core/src/mindustry/input/DesktopInput.java

```diff
@@ -402,6 +403,8 @@ void pollInput(){
         if(Core.input.keyTap(Binding.schematic_menu) && !Core.scene.hasKeyboard()){
             if(ui.schematics.isShown()){
                 ui.schematics.hide();
+            }else if(!state.rules.schematicAllowed) {
+                ui.showInfoToast("@schematic.disabled", 3f);
```

Comment:
People should still be able to save and view schematics, even if the option is disabled. Instead of preventing access to the menu, simply popup a dialog when they click a schematic in the menu instead.

Reply:
I will push a new version with some major changes, so I will close this if it is fine.